table/table.go

```diff
@@ -78,7 +78,7 @@ type Table interface {
 	Truncate(ctx context.Context) (err error)
 
 	// AddRecord inserts a row into the table.
-	AddRecord(ctx context.Context, r []interface{}) (recordID int64, err error)
+	AddRecord(ctx context.Context, r []interface{}, h int64) (recordID int64, err error)
```

Comment:
@qiuyesuifeng 
If we move alloc out from AddRecord, we will have to write `if recordID != 0 { recordID = t.alloc() }` everywhere.
I had also thought about moving all allocs into AddRecord, but found out it not easy either. I will think about it later.